pkg/cmd/init.go

```diff
@@ -0,0 +1,338 @@
+	if err != nil {
+		return err
+	}
+
+	k1config, err := os.OpenFile(filepath.Join(opts.Path, "kubeone.yaml"), os.O_CREATE|os.O_WRONLY, 0600)
+	if err != nil {
+		return fail.Runtime(err, "creating manifest file")
+	}
+	defer k1config.Close()
+
+	ybuf, err := genKubeOneClusterYAML(opts)
+	if err != nil {
+		return fail.Runtime(err, "generating KubeOneCluster")
+	}
+
+	_, err = io.Copy(k1config, bytes.NewBuffer(ybuf))
+	if err != nil {
+		return fail.Runtime(err, "writing KubeOneCluster")
+	}
+
+	if opts.Terraform {
+		prov := validProviders[opts.Provider.String()]
+		if err = examples.CopyTo(opts.Path, prov.terraformPath); err != nil {
+			return fail.Runtime(err, "copying terraform configuration")
+		}
+	}
+
+	return nil
+}
+
+func genKubeOneClusterYAML(opts *initOpts) ([]byte, error) {
+	providerName := opts.Provider.String()
+	prov := validProviders[providerName]
+
+	cluster := kubeonev1beta2.KubeOneCluster{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "KubeOneCluster",
+			APIVersion: kubeonev1beta2.SchemeGroupVersion.Identifier(),
+		},
+		Name: opts.Name,
+		CloudProvider: kubeonev1beta2.CloudProviderSpec{
+			External:    !prov.inTree,
+			CloudConfig: prov.cloudConfig,
+			CSIConfig:   prov.csiConfig,
+		},
+		ContainerRuntime: kubeonev1beta2.ContainerRuntimeConfig{
+			Containerd: &kubeonev1beta2.ContainerRuntimeContainerd{},
+		},
+		Versions: kubeonev1beta2.VersionConfig{
+			Kubernetes: opts.Version,
+		},
+		Addons: &kubeonev1beta2.Addons{
+			Enable: true,
+			Addons: []kubeonev1beta2.Addon{
+				{
+					Name: "default-storage-class",
+				},
+			},
+		},
+	}
+
+	switch strings.Split(providerName, "/")[0] {
+	case "aws":
+		cluster.CloudProvider.AWS = &kubeonev1beta2.AWSSpec{}
+	case "azure":
+		cluster.CloudProvider.Azure = &kubeonev1beta2.AzureSpec{}
+	case "digitalocean":
+		cluster.CloudProvider.DigitalOcean = &kubeonev1beta2.DigitalOceanSpec{}
+	case "equinixmetal":
+		cluster.CloudProvider.DigitalOcean = &kubeonev1beta2.DigitalOceanSpec{}
+	case "gce":
+		cluster.CloudProvider.GCE = &kubeonev1beta2.GCESpec{}
+	case "hetzner":
+		cluster.CloudProvider.Hetzner = &kubeonev1beta2.HetznerSpec{}
+	case "none":
+		cluster.CloudProvider.None = &kubeonev1beta2.NoneSpec{}
+	case "openstack":
+		cluster.CloudProvider.Openstack = &kubeonev1beta2.OpenstackSpec{}
+	case "vmware-cloud-director":
+		cluster.CloudProvider.VMwareCloudDirector = &kubeonev1beta2.VMwareCloudDirectorSpec{}
+	case "vsphere":
+		cluster.CloudProvider.Vsphere = &kubeonev1beta2.VsphereSpec{}
+	default:
+		return nil, fmt.Errorf("unknown provider")
+	}
+
+	buf, err := yaml.Marshal(&cluster)
+	if err != nil {
+		return nil, err
+	}
+
+	doc, err := yamled.Load(bytes.NewBuffer(buf))
+	if err != nil {
+		return nil, err
+	}
+
+	toremove := []string{
+		"apiEndpoint",
+		"clusterNetwork",
+		"controlPlane",
+		"hosts",
+		"features",
+		"loggingConfig",
+		"proxy",
+		"staticWorkers",
+	}
+	for _, yamlPath := range toremove {
+		doc.Remove(yamled.Path{yamlPath})
+	}
```

Comment:
This approach has a disadvantage that if we add some new top-level field to the API (like we did with `loggingConfig`), it's not going to be removed. Can we add some golden file-based test to ensure we catch this?

Reply:
those are removed from the output only for aesthetic reasons, no real impact

Reply:
Yeah, sure, but I'm talking about aesthetics. It's better to omit unnecessary parts as that makes manifest easier to read and understand. And we can easily forget to add some new and not needed field here.

Reply:
done